### ISSUE TITLE
[Feature request] take_along_axis

### ISSUE THREAD

**Issue by naoyam** [NVIDIA/Fuser#204]
https://numpy.org/doc/stable/reference/generated/numpy.take_along_axis.html

Similar to `torch.gather` but less flexible. Unlike `torch.gather`, the input dimensions that are not gathered must have the same sizes as the dimensions of the index tensor, which is not the case with `torch.gather`.

- [x] `take_along_axis` with non fusion inputs (#250)
- [x] Better segmentation for cross entropy. Fuse `take_along_axis` with softmax rather than the final reduction

**Comment by naoyam** [comment]
Closing with #332 . Remaining issues are tracked separately.
